### ISSUE TITLE
Cursor sometimes gets stuck to secondary display

### ISSUE THREAD

**Issue by ocosta** [G-dH/custom-hot-corners-extended#33]
Hi,

I have a notebook as display 1, and an external monitor above it as display 2. The display layout on GNOME reflects this.

When this extension is active, for some unknown reason that I haven't figured out yet (might be related to coming back from lock screen), sometimes the cursor becomes stuck to the secondary display, meaning that it doesn't go beyond the lower boundary of display 2 (where it should enter display 1). Disabling the extension (and restarting GNOME session) fixes the problem.

I am open to ideas on how to help identify the root cause. Is there any log I where can look for some hints?

**Comment by ocosta** [comment]
Hi @G-dH ! Thanks for reaching out.

Sorry for the lousy bug report, my bad. Here's the info you requested:

- Fedora 35, GNOME 41.3, X11, extension v15
- yes, if I recall correctly (will do some additional testing)
- the top left hot corner on the primary display (bottom one) was set to expand 25% vertically
- as far as I remember, simply disabling the extension did not work around the problem

Thanks for the tip on the logs, I will re-enable the extension and take a closer look on how it behaves.

BTW I also have the [Hot Edge](https://github.com/jdoda/hotedge) extension installed, don't know if there could be some sort of conflict

**Comment by G-dH** [comment]
I don't see any conflict between Hot Edge and CHC-E, but it's possible that the blocking barrier actually belongs to the Hot Edge..

**Comment by ocosta** [comment]
That's a possibility, indeed. However, when the cursor got stuck at the bottom edge of the second display (above), it did not trigger GNOME activities -- which is Hot Edge's purpose. Still, there could be something wrong there.
I have already re-enabled CHC-E, will post back here any new info. So far, everything is working as it was supposed to.

**Comment by G-dH** [comment]
You are saying that hitting the barrier didn't trigger Activities, but did it anything at all? These unwanted barriers usually do nothing, I don't know what's their source. CHC-E has option to visualize every barrier and clickable actor that it adds on the screen.

**Comment by ocosta** [comment]
Ok, I did some more testing, and found a way to reproduce de problem:

1. activate screen lock
2. unplug external display (display 2, above primary)
3. unlock the screen
4. re-plug the external monitor

This triggers the "locked in" state for the second display, but only if CHC-E is enabled. Also, I tried disabling CHC-E (and even HotEdge), to no avail.

The only thing that appeared on gnome-shell log was this (not even sure it's CHC-E related):

`st_widget_get_theme_node called on the widget [0x559ac86c90b0 StLabel.monospace:insensitive ("100%")] which is not in the stage.`

I hope it helps you track the problem.

BTW I am using Nvidia's proprietary driver (that's currently the only way I can use an external display -- and only with X11), not sure if it plays any role (I would guess it doesn't, but...)
